services/workflows-service/src/alert/alert.service.ts

```diff
@@ -103,23 +113,132 @@ export class AlertService {
   }
 
   // Function to perform alert checks for each alert definition
-  async checkAllAlerts(): Promise<void> {
+  async checkAllAlerts() {
     const alertDefinitions = await this.getAllAlertDefinitions();
 
     for (const definition of alertDefinitions) {
       const triggered = await this.checkAlert(definition);
+
       if (triggered) {
-        // const ids = await this.createAlert({
-        //   /* ... */
-        // }); // Provide necessary data
+        this.logger.log(`Alert triggered for alert definition ${definition.id}`);
       }
     }
   }
 
   // Specific alert check logic based on the definition
-  private async checkAlert(definition: AlertDefinition): Promise<boolean> {
-    // ...
-    return true;
+  private async checkAlert(alertDefinition: AlertDefinition): Promise<boolean> {
+    const inlineRule = alertDefinition.inlineRule as InlineRule;
+
+    const results = await this.dataAnalyticsService.runInlineRule(
+      alertDefinition.projectId,
+      inlineRule,
+    );
+
+    if (!results || (Array.isArray(results) && results.length === 0)) {
+      return false;
+    }
+
+    // Find the missing keys
+    results.forEach((aggreatedRow: any) => {
+      const missingKeys = _.difference(inlineRule.groupedBy, Object.keys(aggreatedRow));
+      // If there are missing keys, log an error message
+      if (missingKeys.length > 0) {
+        console.error(
+          `Alert aggregated row is missing properties for groupBy: ${missingKeys.join(', ')}`,
+          {
+            inlineRule,
+            aggreatedRow,
+          },
+        );
+      }
+    });
+
+    // const releavntAlerts = await this.getDeduplicatedAlerts(
+    //   alertDefinition.projectId,
+    //   alertDefinition,
+    //   results,
+    // );
```

Comment:
Can we remove this commented out code?

Reply:
its relevant code i just focus on other thing instead the deduplicated logic